administrator/components/com_banners/src/Model/TracksModel.php

```diff
@@ -182,18 +182,18 @@ protected function getListQuery()
     /**
      * Method to delete rows.
      *
-     * @return  boolean  Returns true on success, false on failure.
+     * @return  mixed  Number of deleted tracks on success, false on failure.
      */
     public function delete()
     {
         $user       = $this->getCurrentUser();
         $categoryId = (int) $this->getState('category_id');
 
-        // Access checks.
+        // Access checks. Deleting tracks requires manage permission on the banner or category
         if ($categoryId) {
-            $allow = $user->authorise('core.delete', 'com_banners.category.' . $categoryId);
+            $allow = $user->authorise('core.edit', 'com_banners.category.' . $categoryId);
         } else {
-            $allow = $user->authorise('core.delete', 'com_banners');
+            $allow = $user->authorise('core.edit', 'com_banners');
         }
 
         if ($allow) {
```

Comment:
```suggestion
        if (!$allow) {
            return false;
        }
```
You removed the `else` statement. Do an early return and remove the if statement to reduce indentation.

Reply:
thx, done